File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
answer = 0
def dfs_recursive(idx,numbers,target,value):
    global answer
    n = len(numbers)
    if n == idx and target == value:
        answer += 1
        return
    elif n == idx:
        return
    else:
        dfs_recursive(idx+1,numbers,target,value+numbers[idx])
        dfs_recursive(idx+1,numbers,target,value-numbers[idx])

def solution(numbers, target):
    global answer
    dfs_recursive(0,numbers,target,0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    const dfs = (numbers,idx,curX) =>{
        if(idx == numbers.length && curX == target){
            answer += 1;
            return
        }
        else if(idx == numbers.length && curX != target){
            return;
        }
        
        dfs(numbers, idx + 1, curX - numbers[idx])
        dfs(numbers, idx + 1, curX + numbers[idx]);
    }
    
    dfs(numbers,0,0);
    
    return answer;
}